lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,179 @@ def __init__(self, config):
+        self.re_api_url = config.re_api_url
+        self.version = config.version
+
+    def import_genbank(self, params):
+        print('validating parameters')
+        mass_params = self._set_up_single_params(params)
+        return self._import_genbank_mass(mass_params)[0]
+
+    def import_genbank_mass(self, params):
+        print('validating parameters')
+        self._validate_mass_params(params)
+        return self._import_genbank_mass(params)
+
+    def _set_up_single_params(self, params):
+        # avoid side effects and keep variables in params unmodfied
+        inputs = dict(params)
+        self._validate_params(inputs)
+        ws_id = self._get_int(inputs.pop(_WSID, None), _WSID)
+        ws_name = inputs.pop(_WSNAME, None)
+        if (bool(ws_id) == bool(ws_name)):  # xnor
+            raise ValueError(f"Exactly one of a '{_WSID}' or a '{_WSNAME}' parameter must be provided")
+        if not ws_id:
+            print(f"Translating workspace name {ws_name} to a workspace ID. Prefer submitting "
+                  + "a workspace ID over a mutable workspace name that may cause race conditions")
+            ws_id = self.dfu.ws_name_to_id(ws_name)
+        mass_params = {_WSID: ws_id, _INPUTS: [inputs]}
+        return mass_params
+
+    def _validate_mass_params(self, params):
+        ws_id = self._get_int(params.get(_WSID), _WSID)
+        if not ws_id:
+            raise ValueError(f"{_WSID} is required")
+        inputs = params.get(_INPUTS)
+        if not inputs or type(inputs) != list:
+            raise ValueError(f"{_INPUTS} field is required and must be a non-empty list")
+        for i, inp in enumerate(inputs, start=1):
+            if type(inp) != dict:
+                raise ValueError(f"Entry #{i} in {_INPUTS} field is not a mapping as required")
+            self._validate_params(inp)
+
+    def _get_int(self, putative_int, name, minimum=1):
+        if putative_int is not None:
+            if type(putative_int) != int:
+                raise ValueError(f"{name} must be an integer, got: {putative_int}")
+            if putative_int < minimum:
+                raise ValueError(f"{name} must be an integer >= {minimum}")
+        return putative_int
+
+    def _import_genbank_mass(self, params):
+
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        genome_names = []
+        genome_data = []
+        genome_meta = []
+
+        genome_objs = []
+        consolidated_files = []
+        input_directories = []
+
+        for idx, input_params in enumerate(inputs):
+
+            genome_obj = _Genome(self.cfg)
+            genome_names.append(input_params['genome_name'])
+
+            # construct the input directory staging area
+            input_directory = self._stage_input(input_params)
+            input_directories.append(input_directory)
+
+            # update default params
+            input_params = {**genome_obj.default_params, **input_params}
```

Comment:
Are genome_obj.default_params every modified? if not, since they're populated from the config file, they could be moved from the genome object to a genbank to genome instance variable

Reply:
👍

Reply:
:+1: